File: lib-segment/src/main/java/websiteschema/mpsegment/concept/DuplicateConceptException.java
package websiteschema.mpsegment.concept;

public class DuplicateConceptException extends Exception {

    private Concept existedConcept;

    public DuplicateConceptException(Concept concept) {
        existedConcept = concept;
    }

    @Override
    public String getMessage() {
        return "A concept '" + existedConcept.getName() + "' has id " + existedConcept.getId() + " already existed.";
    }
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/concept/DuplicateConceptException.scala
package websiteschema.mpsegment.concept

class DuplicateConceptException(concept: Concept) extends Exception {

    private val existedConcept : Concept = concept

    override def getMessage() : String = {
        return "A concept '" + existedConcept.getName() + "' has id " + existedConcept.getId() + " already existed."
    }
}
